src/scripts/store.js

```diff
@@ -0,0 +1,319 @@
+  //   country: "Bulgaria",
+  //   photo: require("../img/varna.jpg"),
+  //   visited: true,
+  // },
+  // solun123: {
+  //   id: "solun123",
+  //   city: "Thessaloniki",
+  //   country: "Greece",
+  //   photo: require("../img/thessaloniki.jpg"),
+  //   visited: true,
+  // },
+  // istanbul123: {
+  //   id: "istanbul123",
+  //   city: "Istanbul",
+  //   country: "Turkey",
+  //   photo: require("../img/istanbul.jpg"),
+  //   visited: true,
+  // },
+  // //Wishlist
+  // paris123: {
+  //   id: "paris123",
+  //   city: "Paris",
+  //   country: "France",
+  //   photo: require("../img/paris.jpg"),
+  //   visited: false,
+  // },
+  // santorini123: {
+  //   id: "santorini123",
+  //   city: "Santorini",
+  //   country: "Greece",
+  //   photo: require("../img/santorini.jpg"),
+  //   visited: false,
+  // },
+  // newyork123: {
+  //   id: "newyork123",
+  //   city: "New York",
+  //   country: "U.S.",
+  //   photo: require("../img/newyork.jpg"),
+  //   visited: false,
+  // },
+  // marrakech: {
+  //   id: "marrakech",
+  //   city: "Marrakech",
+  //   country: "Morocco",
+  //   photo: require("../img/marrakech.jpg"),
+  //   visited: false,
+  // },
+  // amsterdam123: {
+  //   id: "amsterdam123",
+  //   city: "Amsterdam",
+  //   country: "Netherlands",
+  //   photo: require("../img/Amsterdam.png"),
+  //   visited: false,
+  // },
+  // berlin123: {
+  //   id: "berlin123",
+  //   city: "Berlin",
+  //   country: "Germany",
+  //   photo: require("../img/berlin.jpg"),
+  //   visited: false,
+  // },
+  // lisbon123: {
+  //   id: "lisbon123",
+  //   city: "Lisbon",
+  //   country: "Porugal",
+  //   photo: require("../img/lisbon.jpg"),
+  //   visited: false,
+  // },
+  // tokyo123: {
+  //   id: "tokyo123",
+  //   city: "Tokyo",
+  //   country: "Japan",
+  //   photo: require("../img/tokyo.jpg"),
+  //   visited: false,
+  // },
+  // sienna123: {
+  //   id: "sienna123",
+  //   city: "Sienna",
+  //   country: "Italy",
+  //   photo: require("../img/sienna.jpg"),
+  //   visited: false,
+  // },
+  // dublin123: {
+  //   id: "dublin123",
+  //   city: "Dublin",
+  //   country: "Ireland",
+  //   photo: require("../img/dublin.jpg"),
+  //   visited: false,
+  // },
+  // petersburg123: {
+  //   id: "petersburg123",
+  //   city: "Saint petersburg",
+  //   country: "Russia",
+  //   photo: require("../img/saintpetersburg.jpg"),
+  //   visited: false,
+  // },
+};
+
+export function getStoredList() {
+  const storedListLS = localStorage.getItem("destinations");
+
+  let storedList = storedListLS ? JSON.parse(storedListLS) : [];
+  return storedList;
+}
+
+let storedList = getStoredList();
+
+let storedKeys = storedList.map((element) => {
+  return element.id;
+});
+
+export function toggleButtonVisited(event, item) {
+  var checkbox = event.target;
+  if (checkbox.checked) {
+    checkbox.parentElement.classList.add("visited");
+    checkbox.parentElement.classList.remove("bucketlist");
+  }
```

Comment:
Remember the MVC video I shared with you? Only the "View" part cares about the layout itself. That means in here shouldn't happen mutations to the DOM. The store is only responsible for data flow. So what should happen here is to update the LS item visited status.

Remember, that now that the LS is an object, you don't need to loop it, you can just directly change it  `destinationsObj[itemId].visited = true|false`;

The way you mutate the LS is very similar to how we mutate it when adding a new item, but instead of adding a new item, we change an existing one. Good luck!